client/rpc-servers/src/lib.rs

```diff
@@ -81,48 +104,63 @@ mod inner {
 						"version": 1,
 						"methods": methods,
 					}))
-				}).unwrap();
+				}).expect("infallible all other methods have their own address space; qed");
 
 				server.register_module(methods_api).unwrap();
+				let _ = tx.send(Ok(handle));
 				let _ = server.start().await;
 			});
 		});
 
-		Ok(())
+		rx.await.unwrap_or(Err("Channel closed".to_string()))
```

Comment:
Should this be returning a custom Error enum rather than String as error?

Reply:
yes, ideally I think we could use our internal error or just `anyhow:Error` but for now it's all right I think.